TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/EditorTests/PropertyTests/UnityTexture2DPropertyTests.cs

```diff
@@ -0,0 +1,81 @@
+using System.Collections;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace UnityEditor.ShaderFoundry.UnitTests
+{
+    class UnityTexture2DPropertyTests : BlockTestRenderer
+    {
+        UnityTexture2DPropertyBlockBuilder BuildWithoutNameOverrides(string defaultValue)
+        {
+            var propBuilder = new UnityTexture2DPropertyBlockBuilder
+            {
+                PropertyAttribute = new PropertyAttributeData { DefaultValue = defaultValue },
+            };
+            return propBuilder;
+        }
+
+        UnityTexture2DPropertyBlockBuilder BuildWithNameOverrides(string defaultValue)
+        {
+            var propBuilder = new UnityTexture2DPropertyBlockBuilder
+            {
+                PropertyAttribute = new PropertyAttributeData
+                {
+                    UniformName = "_Texture",
+                    DisplayName = "Texture",
+                    DefaultValue = defaultValue
+                },
+            };
+            return propBuilder;
+        }
+
+        [UnityTest]
+        public IEnumerator Texture2DProperty_DefaultPropertyIsWhite_ExpectedColorIsWhite()
+        {
+            var expectedColor = new Color(1, 1, 1, 1);
+
+            var container = CreateContainer();
+            var propBuilder = BuildWithoutNameOverrides("\"white\" {}");
+            var block = propBuilder.Build(container);
+
+            TestSurfaceBlockIsConstantColor(container, propBuilder.BlockName, block, expectedColor);
+
+            yield break;
+        }
+
+        [UnityTest]
+        public IEnumerator Texture2DProperty_DefaultPropertyIsBlack_ExpectedColorIsBlack()
+        {
+            var expectedColor = new Color(0, 0, 0, 0);
+
+            var container = CreateContainer();
+            var propBuilder = BuildWithoutNameOverrides("\"black\" {}");
+            var block = propBuilder.Build(container);
+
+            TestSurfaceBlockIsConstantColor(container, propBuilder.BlockName, block, expectedColor);
+
+            yield break;
+        }
+
+        [UnityTest]
+        public IEnumerator Texture2DProperty_MaterialColorSet_IsExpectedColor()
+        {
+            var expectedColor = new Color(0.24f, 0.49f, 0.74f, 1.0f);
```

Comment:
Is there a particular reason to use 0.24, 0.49, 0.74?
Cubemap test uses 0.25, 0.5, 0.75 :)

Reply:
`TestSurfaceBlockIsConstantColor` takes a Color32. It looks like this gets auto converted. I used Color so it matched what I was setting so the test was more readable. I'd have to go do some plumbing in the testbed to convert it from Color32 to Color internally.